### PR TITLE
Added documentation about mixins

### PR DESCRIPTION
**Pull request by rmadsen-ks**

### CODE REVIEW

File: doc/Developer Guide/Test Step/Readme.md [hunk 1]
```diff
@@ -627,3 +627,103 @@ Remember, adhering to the correct syntax is crucial for the expressions to evalu
 
    Example: `The number is {1 + 2}.` becomes "The number is 3."
 
+## Mixins
+
+Mixins, short for "mix-ins," are small units of functionality that can be integrated or 'mixed in' with an object.
```
Comment (bdirenzo-keysight) [inline doc/Developer Guide/Test Step/Readme.md:632]: I think this uses the word MixIn to describe what a MixIn is too frequently. Maybe something like. 
The term MixIn comes from object-oriented programming where a mix-in is a class that contains extensions for use by other classes.

File: doc/Developer Guide/Test Step/Readme.md [hunk 2]
```diff
@@ -627,3 +627,103 @@ Remember, adhering to the correct syntax is crucial for the expressions to evalu
 
    Example: `The number is {1 + 2}.` becomes "The number is 3."
 
+## Mixins
+
+Mixins, short for "mix-ins," are small units of functionality that can be integrated or 'mixed in' with an object. 
+In the context of OpenTAP, mixins are can be used for extending the capabilities of a test step.
```
Comment (bdirenzo-keysight) [inline doc/Developer Guide/Test Step/Readme.md:633]: Don't we also support DUTs, Instruments, and Result Listeners? At least the prompt comes up for those

File: doc/Developer Guide/Test Step/Readme.md [hunk 3]
```diff
@@ -627,3 +627,103 @@ Remember, adhering to the correct syntax is crucial for the expressions to evalu
...
+
+### Getting Started
+
+By default, OpenTAP contains no mixins, but they can be added with packages. Notable packages includes:
+- Expressions: Adds Number and Text mixins.
+- Basic Mixins: Contains a number of basic mixins. For example the Repeat mixins, which enables a test step to repeat itself. Or Artifact which allows adding a file path pointing to a file that should be considered an artifact of the step.
```
Comment (bdirenzo-keysight) [inline doc/Developer Guide/Test Step/Readme.md:639]: And Limits to apply a limit check based on a value calculated in the test step

File: doc/Developer Guide/Test Step/Readme.md [hunk 4]
```diff
@@ -627,3 +627,103 @@ Remember, adhering to the correct syntax is crucial for the expressions to evalu
...
+
+### Adding Mixins as an End User
+
+The mixins are available through the settings context menu. Right-click a setting or in the settings area and click "Add Mixin".
```
Comment (bdirenzo-keysight) [inline doc/Developer Guide/Test Step/Readme.md:643]: Should this part (+ some examples/screenshots) be in the 8400 documentation instead?

**Review by bdirenzo-keysight — CHANGES_REQUESTED** [review]

File: doc/Developer Guide/Test Step/Readme.md (hunk 2 above)
Comment (rmadsen-ks) [reply doc/Developer Guide/Test Step/Readme.md]: Ill add test step as an example. We still dont have any mixins for those types.

File: doc/Developer Guide/Test Step/Readme.md (hunk 4 above)
Comment (rmadsen-ks) [reply doc/Developer Guide/Test Step/Readme.md:645]: I think it is generic enough to have as a paragraph here. I should have more info in the KS8400 documentation.

**Pull request merged**